network/gossip/libp2p/libp2pNode_test.go

```diff
@@ -332,75 +329,62 @@ func (l *LibP2PNodeTestSuite) TestCreateStreamTimeoutWithUnresponsiveNode() {
 	// attempt to create a stream from node 1 to node 2 and assert that it fails after timeout
 	grace := 10 * time.Millisecond
 	var err error
-	unittest.AssertReturnsBefore(l.T(),
+	unittest.AssertReturnsBefore(suite.T(),
 		func() {
 			_, err = peers[0].CreateStream(ctx, silentNodeAddress)
 		},
 		DefaultUnicastTimeout+grace)
-	assert.Error(l.T(), err)
+	assert.Error(suite.T(), err)
 }
 
 // TestCreateStreamIsConcurrent tests that CreateStream calls can be made concurrently such that one blocked call
 // does not block another concurrent call.
-func (l *LibP2PNodeTestSuite) TestCreateStreamIsConcurrent() {
+func (suite *LibP2PNodeTestSuite) TestCreateStreamIsConcurrent() {
 
 	// bump up the unicast timeout to a high value
 	unicastTimeout = time.Hour
 
 	// create two regular node
-	goodPeers, goodAddrs := l.CreateNodes(2, nil, false)
-	defer l.StopNodes(goodPeers)
-	require.Len(l.T(), goodAddrs, 2)
+	goodPeers, goodAddrs := suite.CreateNodes(2, nil, false)
+	defer suite.StopNodes(goodPeers)
+	require.Len(suite.T(), goodAddrs, 2)
 
 	// create a silent node which never replies
-	listener, silentNodeAddress := newSilentNode(l.T())
-	defer listener.Close()
-
-	wg := sync.WaitGroup{}
-	wg.Add(1)
-
-	ch := make(chan struct{})
-	// spin off a go routine to create a stream to the unresponsive node
-	go func() {
-		wg.Done()
-		_, _ = goodPeers[0].CreateStream(l.ctx, silentNodeAddress) // this call will block
-		close(ch)
+	listener, silentNodeAddress := newSilentNode(suite.T())
+	defer func() {
```

Comment:
wouldn't this work instead? `defer require.NoError(suite.T(), listener.Close())`

Reply:
thanks, fixed in https://github.com/onflow/flow-go/pull/99/commits/cc27eb3e89012ebc7b57c46ffcbc040def7d29f5.